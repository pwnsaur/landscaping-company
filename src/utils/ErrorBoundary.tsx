import React, { ErrorInfo } from 'react';

interface Props {
  children: React.ReactNode;
}

interface State {
  hasError: boolean;
}

class ErrorBoundary extends React.Component<Props, State> {
  constructor(props: Props) {
    super(props);

    this.state = { hasError: false };
  }

  static getDerivedStateFromError(_: Error): State {
    return { hasError: true };
  }

  componentDidCatch(error: Error, errorInfo: ErrorInfo) {
    console.log({ error, errorInfo });
  }

  render() {
    if (this.state.hasError) {
      return (
        <div>
          <h1>Lapu neizdevās ielādēt.</h1>
        </div>
      );
    }

    return this.props.children;
  }
}

export default ErrorBoundary;
