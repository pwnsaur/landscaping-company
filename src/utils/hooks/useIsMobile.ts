import useWindowSize from './useWindowSize';

const useIsMobile = (): boolean => {
  const { width, height } = useWindowSize();
  const currentWidth = width || 0;
  const currentHeight = height || 0;
  const isMobile = currentWidth <= 640;

  return isMobile;
};

export default useIsMobile;
