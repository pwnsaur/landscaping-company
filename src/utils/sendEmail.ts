import axios, { AxiosResponse } from 'axios';

import { FormData } from '@/types/contactForm';

type ServerResponse = {
  messageId: string;
  response: string;
};

type SendEmailFn = (
  formData: FormData
) => Promise<AxiosResponse<ServerResponse>>;

const sendEmail: SendEmailFn = async (formData: FormData) => {
  return axios({
    method: 'post',
    url: '/api/send-mail',
    data: formData,
  });
};

export default sendEmail;
