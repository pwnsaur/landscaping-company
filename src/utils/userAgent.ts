export function isMobileUserAgent(userAgent: string): boolean {
  return /Android|iPhone|iPod|BlackBerry|IEMobile|Opera Mini/i.test(userAgent);
}
