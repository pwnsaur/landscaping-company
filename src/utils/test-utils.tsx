import { ReactElement, ReactNode } from 'react';
import { ThemeProvider } from 'styled-components';

import { theme } from '@/styles/theme';
import useIsMobile from '@/utils/hooks/useIsMobile';
import { render, RenderOptions } from '@testing-library/react';

interface TestWrapperProps {
  children: ReactNode;
}

const TestWrapper = ({ children }: TestWrapperProps) => {
  const isMobile = useIsMobile();

  return (
    <ThemeProvider theme={{ ...theme, isMobile }}>{children}</ThemeProvider>
  );
};

const customRender = (
  ui: ReactElement,
  options?: Omit<RenderOptions, 'queries'>
) => {
  return render(ui, { wrapper: TestWrapper, ...options });
};

// eslint-disable-next-line import/export
export * from '@testing-library/react';

// eslint-disable-next-line import/export
export { customRender as render };
