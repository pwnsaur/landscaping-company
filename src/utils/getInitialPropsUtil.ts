import { AppContext, AppInitialProps } from 'next/app';

import { isMobileUserAgent } from './userAgent';

export async function getInitialPropsUtil(
  appContext: AppContext
): Promise<AppInitialProps & { isMobile: boolean }> {
  let pageProps = {};

  if (appContext.Component.getInitialProps) {
    pageProps = await appContext.Component.getInitialProps(appContext.ctx);
  }

  const userAgent = appContext.ctx.req?.headers['user-agent'] || 'rekt';
  const isMobile = isMobileUserAgent(userAgent);

  return { pageProps, isMobile };
}
