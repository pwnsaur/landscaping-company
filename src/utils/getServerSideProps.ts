import { GetServerSideProps } from 'next';

import { isMobileUserAgent } from '@utils/userAgent';

export const getServerSideProps: GetServerSideProps = async ({ req }) => {
  const userAgent = req.headers['user-agent'] || 'about rekt';
  const isMobile = isMobileUserAgent(userAgent);
  return { props: { isMobile } };
};
