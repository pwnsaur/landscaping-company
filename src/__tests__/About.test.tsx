import About from '@/pages/about';
import useIsMobile from '@/utils/hooks/useIsMobile';
import { render } from '@/utils/test-utils';

jest.mock('../utils/hooks/useIsMobile');

describe('About Page', () => {
  test('matches the snapshot in desktop view', () => {
    (useIsMobile as jest.Mock).mockReturnValue(false);
    const { asFragment } = render(<About />);
    expect(asFragment()).toMatchSnapshot();
  });

  test('matches the snapshot in mobile view', () => {
    (useIsMobile as jest.Mock).mockReturnValue(true);
    const { asFragment } = render(<About />);
    expect(asFragment()).toMatchSnapshot();
  });
});
