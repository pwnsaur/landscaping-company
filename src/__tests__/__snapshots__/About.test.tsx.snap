// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`About Page matches the snapshot in desktop view 1`] = `
<DocumentFragment>
  .c0 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-box-pack: justify;
  -webkit-justify-content: space-between;
  -ms-flex-pack: justify;
  justify-content: space-between;
  margin: 4em;
}

.c1 {
  min-width: 40%;
  -webkit-align-self: center;
  -ms-flex-item-align: center;
  align-self: center;
}

.c3 {
  max-width: 60%;
  padding: 0 4rem;
}

.c4 {
  font-size: 1.2rem;
  text-transform: uppercase;
  margin-bottom: 1.25rem;
}

.c5 {
  font-size: 1rem;
  line-height: 1.5;
  margin-bottom: 1.25rem;
}

.c2 {
  width: 100%;
  height: 100%;
  object-fit: cover;
}

<div
    class="c0"
  >
    <div
      class="c1"
    >
      <img
        alt="cato"
        class="c2"
        data-nimg="1"
        decoding="async"
        height="2688"
        src="/_next/image?url=%2Fimg.jpg&w=3840&q=40"
        srcset="/_next/image?url=%2Fimg.jpg&w=1920&q=40 1x, /_next/image?url=%2Fimg.jpg&w=3840&q=40 2x"
        style="color: transparent;"
        width="1920"
      />
    </div>
    <div
      class="c3"
    >
      <h2
        class="c4"
      >
        Hey it's us
      </h2>
      <p
        class="c5"
      >
        Lorem ipsum dolor sit amet, consectetur adipiscing elit. Proin sit amet purus ac eros luctus dictum vel id dui. Nunc suscipit est vel augue euismod, quis tristique mauris luctus. Integer fringilla elit a eros mollis, nec dignissim eros efficitur. Sed eget eleifend lorem. Sed non arcu ac tortor consectetur lobortis vel vel arcu.
      </p>
    </div>
  </div>
</DocumentFragment>
`;

exports[`About Page matches the snapshot in mobile view 1`] = `
<DocumentFragment>
  .c0 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-box-pack: justify;
  -webkit-justify-content: space-between;
  -ms-flex-pack: justify;
  justify-content: space-between;
  margin: 4em;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
  margin: 0.5rem;
  margin: 5rem 1rem 2rem;
}

.c1 {
  min-width: 40%;
  -webkit-align-self: center;
  -ms-flex-item-align: center;
  align-self: center;
  max-width: 70%;
}

.c3 {
  max-width: 60%;
  padding: 0 4rem;
  max-width: 100%;
  padding: 1rem;
  text-align: center;
}

.c4 {
  font-size: 1.2rem;
  text-transform: uppercase;
  margin-bottom: 1.25rem;
}

.c5 {
  font-size: 1rem;
  line-height: 1.5;
  margin-bottom: 1.25rem;
}

.c2 {
  width: 100%;
  height: 100%;
  object-fit: cover;
}

<div
    class="c0"
  >
    <div
      class="c1"
    >
      <img
        alt="cato"
        class="c2"
        data-nimg="1"
        decoding="async"
        height="2688"
        src="/_next/image?url=%2Fimg.jpg&w=3840&q=40"
        srcset="/_next/image?url=%2Fimg.jpg&w=1920&q=40 1x, /_next/image?url=%2Fimg.jpg&w=3840&q=40 2x"
        style="color: transparent;"
        width="1920"
      />
    </div>
    <div
      class="c3"
    >
      <h2
        class="c4"
      >
        Hey it's us
      </h2>
      <p
        class="c5"
      >
        Lorem ipsum dolor sit amet, consectetur adipiscing elit. Proin sit amet purus ac eros luctus dictum vel id dui. Nunc suscipit est vel augue euismod, quis tristique mauris luctus. Integer fringilla elit a eros mollis, nec dignissim eros efficitur. Sed eget eleifend lorem. Sed non arcu ac tortor consectetur lobortis vel vel arcu.
      </p>
    </div>
  </div>
</DocumentFragment>
`;
